src/components/intro/StudioToon.tsx

```diff
@@ -0,0 +1,86 @@
+import React, {useEffect, useState} from "react";
+
+import SectionTitle from "@components/commons/SectionTitle";
+
+import {
+    StudioToonBox,
+    StudioToonWrapper,
+    SubTitle,
+    ToonDetail,
+    ToonItemWrapper,
+    ToonItem,
+    ToonInfo,
+    ToonVisual
+} from "@styles/intro/index.style";
+import Modal from "@components/commons/Modal";
+
+
+function StudioToon({ scrollY }: { scrollY: number }) {
+
+    const [childrenCount, setChildrenCount] = useState(0);
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [itemId, setItemId] = useState(0);
+
+    useEffect(() => {
+        const toonItemWrapper = document.querySelector("#toon-item-wrapper");
+        if (toonItemWrapper) {
+            const count = toonItemWrapper.children.length;
+            setChildrenCount(count);
+            console.log("childrenCount: ", count);
+        }
+    }, []);
+
+    function modalOpen(itemNumber: number, ) {
+        if(isModalOpen) {
+            setIsModalOpen(false);
+        } else {
+            setIsModalOpen(true);
+            setItemId(itemNumber);
+        }
+    }
```

Comment:
```suggestion
// hooks/modal.hook.ts
import { useState } from "react";

export const useModal = () => {
  const [isModalOpen, setIsModalOpen] = useState(false);
  
  const toggleModal = (e: React.MouseEvent) => {
    e.preventDefault();
    e.stopPropagation();
    
    if (isModalOpen) setIsModalOpen(false);
    else {
      setIsModalOpen(true);
      setItemId(itemNumber);
    }
  };
  
  return {
    isModalOpen,
    toggleModal
  };
};

// components/intro/StudioToon.tsx
...
  const { isModalOpen, toggleModal } = useModal();
...
```
함수명은 `openModal`이지만 `Modal`을 닫는 로직도 포함하고 있습니다. 변수명을 `toggleModal`로 수정하거나 `openModal`, `closeModal`로 로직을 분리했으면 좋겠습니다.

또한, Modal 컴포넌트의 재사용성을 높이기 위해 관련 값들은 hook으로 분리하는게 좋을 것 같습니다.